middleware/proxy/reverseproxy.go

```diff
@@ -58,6 +58,12 @@ func singleJoiningSlash(a, b string) string {
 	return a + b
 }
 
+func socketDial(hostName string) func(network, addr string) (conn net.Conn, err error) {
+	return func(network, addr string) (conn net.Conn, err error) {
+		return net.Dial("unix", hostName[len("unix://"):])
```

Comment:
Below, you're checking for a "unix:" prefix - I suppose that a url parse somewhere converts it so the prefix is "unix://" ?